src/js/main.js

```diff
@@ -1,7 +1,451 @@
+      },
+      properties: {
+        Id: 16,
+        Location: 'Fish Eagle',
+        Date: '8/8/2018',
+        Day: 'Wednesday',
+        Time: '2:29:37 PM',
+        'Time of Day': 'PM',
+        'Tidal State': 'High',
+        'Spring/Neap': '',
+        'Wav-File-Name': 'Ish_Bomb20180808_142936_784',
+        'Blast Bearing': 15.38,
+        'Detection Radius(km)': '30'
+      }
+    },
+    {
+      type: 'Feature',
+      geometry: {
+        type: 'Point',
+        coordinates: [39.30479, -4.86934]
+      },
+      properties: {
+        Id: 17,
+        Location: 'Fish Eagle',
+        Date: '8/21/2018',
+        Day: 'Tuesday',
+        Time: '3:20:55 PM',
+        'Time of Day': 'PM',
+        'Tidal State': 'Falling',
+        'Spring/Neap': '',
+        'Wav-File-Name': 'Ish_Bomb20180821_152054_827',
+        'Blast Bearing': 15.41,
+        'Detection Radius(km)': '30'
+      }
+    },
+    {
+      type: 'Feature',
+      geometry: {
+        type: 'Point',
+        coordinates: [39.07585, -5.49254]
+      },
+      properties: {
+        Id: 18,
+        Location: 'Maziwe Island',
+        Date: '9/29/2018 9:35:14',
+        Day: 'Saturday',
+        Time: '09:35:14AM',
+        'Time of Day': 'AM',
+        'Tidal State': '',
+        'Spring/Neap': '',
+        'Wav-File-Name': 'Ish_Bomb20180929_093514_719',
+        'Blast Bearing': 17.54,
+        'Detection Radius(km)': '30'
+      }
+    }
+  ]
+};
+
+//get geojson coordinates value as center of circle
+let coordinateValues = [];
+Object.keys(blastData.features).forEach(e => {
+  coordinateValues.push(blastData.features[e].geometry.coordinates);
+});
+
+function geoJSONCircleRadius(center, radiusInKm, points) {
+  if (!points) points = 64;
+
+  const coords = {
+    latitude: center[1],
+    longitude: center[0]
+  };
+  const km = radiusInKm;
+
+  const ret = [];
+  const distanceX = km / (111.32 * Math.cos((coords.latitude * Math.PI) / 180));
+  const distanceY = km / 110.574;
+  for (var i = 0; i < points; i++) {
+    theta = (i / points) * (2 * Math.PI);
+    x = distanceX * Math.cos(theta);
+    y = distanceY * Math.sin(theta);
+
+    ret.push([coords.longitude + x, coords.latitude + y]);
+  }
+  ret.push(ret[0]);
+
+  return {
+    type: 'geojson',
+    data: {
+      type: 'FeatureCollection',
+      features: [
+        {
+          type: 'Feature',
+          geometry: {
+            type: 'Polygon',
+            coordinates: [ret]
+          }
+        }
+      ]
+    }
+  };
+}
+
+map.on('load', function() {
+  map.addLayer({
+    id: 'points',
+    type: 'circle',
+    source: {
+      type: 'geojson',
+      data: blastData
+    },
+    layout: {},
+    paint: {
+      'circle-color': 'red',
+      'circle-opacity': 0.8
+    }
+  });
+
+  //map.addSource('polygon', geoJSONCircleRadius([39.07585, -5.49254], 30));
+
+  coordinateValues.forEach(function(index, value) {
+    map.addSource('polygon-' + index, geoJSONCircleRadius(value, 30));
```

Comment:
only create one Source and Layer, but include all blast radius features in a single GeoJSON FeatureCollection